src/test/java/jp/co/yahoo/yosegi/util/TestRangeBinarySearch.java

```diff
@@ -0,0 +1,183 @@
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.Arguments;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+import java.util.List;
+
+public class TestRangeBinarySearch {
+
+  @Test
+  public void T_newInstace_1() {
+    RangeBinarySearch<String> bs = new RangeBinarySearch<String>();
+  }
+
+  @Test
+  public void T_add_get_1() {
+    RangeBinarySearch<String> bs = new RangeBinarySearch<String>();
+    bs.add( "test0" , 0 );
+    assertEquals( bs.get(0) , "test0" );
+  }
+
+  @Test
+  public void T_add_get_2() {
+    RangeBinarySearch<String> bs = new RangeBinarySearch<String>();
+    bs.add( "test5" , 5 );
+    assertEquals( bs.get(5) , "test5" );
+  }
+
+  @Test
+  public void T_add_get_3() {
+    RangeBinarySearch<String> bs = new RangeBinarySearch<String>();
+    bs.add( "test0" , 0 );
+    bs.add( "test1" , 1 );
+    bs.add( "test2" , 2 );
+    bs.add( "test3" , 3 );
+    bs.add( "test4" , 4 );
+    bs.add( "test5" , 5 );
+    assertEquals( bs.get(0) , "test0" );
+    assertEquals( bs.get(1) , "test1" );
+    assertEquals( bs.get(2) , "test2" );
+    assertEquals( bs.get(3) , "test3" );
+    assertEquals( bs.get(4) , "test4" );
+    assertEquals( bs.get(5) , "test5" );
+    assertEquals( bs.get(6) , null );
+  }
+
+  @Test
+  public void T_add_get_4() {
+    RangeBinarySearch<String> bs = new RangeBinarySearch<String>();
+    bs.add( "test5" , 5 );
+    assertEquals( bs.get(0) , null );
+    assertEquals( bs.get(1) , null );
+    assertEquals( bs.get(2) , null );
+    assertEquals( bs.get(3) , null );
+    assertEquals( bs.get(4) , null );
+    assertEquals( bs.get(5) , "test5" );
+    assertEquals( bs.get(6) , null );
+  }
+
+  @Test
+  public void T_add_get_5() {
+    RangeBinarySearch<String> bs = new RangeBinarySearch<String>();
+    bs.add( "test0" , 0 );
+    bs.add( "test5" , 5 );
+    assertEquals( bs.get(0) , "test0" );
+    assertEquals( bs.get(1) , null );
+    assertEquals( bs.get(2) , null );
+    assertEquals( bs.get(3) , null );
+    assertEquals( bs.get(4) , null );
+    assertEquals( bs.get(5) , "test5" );
+    assertEquals( bs.get(6) , null );
+  }
+
+  @Test
+  public void T_add_get_6() {
+    RangeBinarySearch<String> bs = new RangeBinarySearch<String>();
+    assertThrows( RuntimeException.class ,
+      () -> {
+        bs.add( "test0" , -1 );
+      }
+    );
+  }
+
+  @Test
+  public void T_add_get_7() {
+    RangeBinarySearch<String> bs = new RangeBinarySearch<String>();
+    bs.add( "test0" , 0 );
+    assertThrows( RuntimeException.class ,
+      () -> {
+        bs.add( "test0" , 0 );
+      }
+    );
+  }
+
+  @Test
+  public void T_add_get_8() {
+    RangeBinarySearch<String> bs = new RangeBinarySearch<String>();
+    bs.add( "test0" , 1 );
+    assertThrows( RuntimeException.class ,
+      () -> {
+        bs.add( "test0" , 0 );
+      }
+    );
+  }
+
```

Comment:
Please describe what are tested in the test method separately at T_add_get_?.
For example, T_add_get_when_xxx or some.

Reply:
Please add a prefix or suffix like T_ or Text or some so that the code reader knows it is a test method including assertThat or assertThrows and/or some testing methods.